e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
This, perhaps? 

``` C
#ifdef NDEBUG
# define alt_assert(x) (x) 
#else
# define alt_assert(x) ((int)assert((x)), 1)
#endif
```

I'm unsure `assert` can be treated that way, but I'm sure we can come up with a workable alternative.